src/modules/consumer.js

```diff
@@ -21,36 +21,29 @@ class Consumer {
 
   /**
    * Get a function to execute on incoming messages to handle RPC
-   * @param  {any} msg   An amqp.node message object
+   * @param  {any} messageProperties   An amqp.node message properties object, containing the rpc settings
    * @param  {string} queue The initial queue on which the handler received the message
-   * @return {function}       a function to use in a chaining on incoming messages
+   * @param  {any} reply the received message to reply the rpc if needed:
+   * @return {any}       object, string, number... the current received message
    */
-  checkRpc(msg, queue) {
-    /**
-     * When message contains a replyTo property, we try to send the answer back
-     * @param  {any} content the received message:
-     * @return {any}          object, string, number... the current received message
-     */
-    return (content) => {
-      if (msg.properties.replyTo) {
-        const options = {
-          correlationId: msg.properties.correlationId,
-          persistent: true,
-          durable: true,
-        };
-        this._connection.config.transport.debug(loggerAlias, `[${queue}][${msg.properties.replyTo}] >`, content);
-        this._connection.config.logger.debug({
-          message: `${loggerAlias} [${queue}][${msg.properties.replyTo}] > ${content}`,
-          params: { content },
-        });
+  async checkRpc(messageProperties, queue, reply) {
+    if (messageProperties.replyTo) {
+      const options = {
+        correlationId: messageProperties.correlationId,
+        persistent: true,
+        durable: true,
+      };
+      this._connection.config.transport.debug(loggerAlias, `[${queue}][${messageProperties.replyTo}] >`, reply);
+      this._connection.config.logger.debug({
+        message: `${loggerAlias} [${queue}][${messageProperties.replyTo}] > ${reply}`,
+        params: { content: reply },
+      });
 
-        return this._connection.getDefaultChannel().then((channel) => {
-          return channel.sendToQueue(msg.properties.replyTo, parsers.out(content, options), options);
-        });
-      }
+      const defaultChannel = await this._connection.getDefaultChannel();
+      return defaultChannel.sendToQueue(messageProperties.replyTo, parsers.out(reply, options), options);
```

Comment:
Please use `return await` where applicable, it preserves the stack-trace on async functions

Reply:
That's against the project's eslint rules. If I do that there is this error when linting:
> Redundant use of `await` on a return value  no-return-await